thunder/core/jit_ext.py

```diff
@@ -1500,12 +1500,12 @@ def process_recorded_modifications(ctx, epilogue_trace):
 
 def bind_inputs(name, trace, input_vars, input_proxies):
     # Unpacks inputs into the computation trace
-    # TODO This currently does the unpacks at the end of he trace, then moves them to the beginning, there's
+    # TODO This currently does the unpacks at the end of the trace, then moves them to the beginning, there's
     #   almost certainly a more elegant way to do this
     with tracectx(trace):
         p: Proxy
         for p in input_proxies:
-            prims.unpack_trivial(p)
+            prims.unpack_trivial(p, name=p.name)
```

Comment:
Would it make sense to fail when name is not given?

Reply:
It might! Because without name it's hard to identify the input at later stages. Should the check mechanism be located in the implementation of the prim?

Reply:
Anything that errors with the call to the prim in the backtrace... I think bind_postprocess might be a good location.

Reply:
`bind_postprocess` is a callback that is called after the bound symbol is created. The backtrace might not be that nice, I think Riccardo's idea of putting the check in the implementation or meta of the prim would work better and more consistent with all other checks in the codebase, I don't think we have any errors put into bind_postprocess today.

Reply:
That's a very interesting bind_postprocess:
https://github.com/Lightning-AI/lightning-thunder/blob/20aa0bd51cfbf9c9786bf12b492268cc5b721c79/thunder/core/prims.py#L659-L661
We call this Symbol with some proxies that are then put in the BoundSymbol as output and its args are cleared. What would be a less special way to do this?

Reply:
Why do we need unpack_trivial at all?

Reply:
> bind_postprocess is a callback that is called after the bound symbol is created. The backtrace might not be that nice, I think Riccardo's idea of putting the check in the implementation or meta of the prim would work better and more consistent with all other checks in the codebase, I don't think we have any errors put into bind_postprocess today.

True! Thank you.

> that's a very interesting bind_postprocess:
maybe with requiring the name, we could drop this and just create it with empty inputs.

> Why do we need unpack_trivial at all?

We use `unpack_trivial` to let every proxy have a creator. I'm not attached to it but it seems a reasonable invariant.

Reply:
But so for this patch, should we add a check to the meta and merge?